source/guide_rocketchat.rst

```diff
@@ -0,0 +1,305 @@
+::
+
+ [isabell@stardust ~]$ mongo --username <username>_mongoroot --eval "printjson(rs.initiate())"
+ MongoDB shell version v4.2.3
+ Enter password: 
+ connecting to: mongodb://127.0.0.1:27017/?compressors=disabled&gssapiServiceName=mongodb
+ Implicit session: session { "id" : UUID("d0f22c61-382c-4178-9a6d-09b42674c14d") }
+ MongoDB server version: 4.2.3
+ {
+ 	"info2" : "no configuration specified. Using a default configuration for the set",
+ 	"me" : "127.0.0.1:27017",
+ 	"ok" : 1
+ }
+ [isabell@stardust ~]$
+
+.. note:: You don't need to add ``--username`` if you have stored your credentials in the `~/.mongorc.js` file.
+ But if you add it then don't forget to replace ``<username>`` with your username.
+
+As last part of the MongoDB configuration we need a user for Rocket.Chat.
+Generate a random password:
+
+::
+
+ [isabell@stardust ~]$ < /dev/urandom tr -dc _A-Z-a-z-0-9 | head -c${1:-15};echo;
+ randompassword
+ [isabell@stardust ~]$
+
+Create ``~/rocket.chat-user-setup.js`` with the following content:
+
+.. code-block:: none
+ :emphasize-lines: 3,4
+
+ db.createUser(
+    {
+      user: "<username>_rocketchat",
+      pwd: "<password>",
+      roles: [
+        {role:"dbOwner", db:"rocketchat"},
+        {role:"readWrite", db:"local"},
+        {role:"clusterMonitor", db:"admin"}
+      ]
+    }
+ )
+
+Replace ``<username>`` with your username and ``<password>`` with the generated password.
+
+Use mongo to add the user:
+
+::
+
+ [isabell@stardust ~]$ mongo admin --username <username>_mongoroot ~/rocket.chat-user-setup.js
+ MongoDB shell version v4.2.3
+ Enter password: 
+ connecting to: mongodb://127.0.0.1:27017/admin?compressors=disabled&gssapiServiceName=mongodb
+ Implicit session: session { "id" : UUID("de28d438-1b38-4c59-964b-8a2f7f88d2d9") }
+ MongoDB server version: 4.2.3
+ Successfully added user: {
+	 "user" : "isabell_rocketchat",
+	 "roles" : [
+ 		{
+ 			"role" : "dbOwner",
+ 			"db" : "rocketchat"
+ 		},
+ 		{
+ 			"role" : "readWrite",
+ 			"db" : "local"
+ 		},
+ 		{
+ 			"role" : "clusterMonitor",
+ 			"db" : "admin"
+ 		}
+ 	]
+ }
+ [isabell@stardust ~]$ 
+
+.. note:: You don't need to add ``--username`` if you have stored your credentials in the `~/.mongorc.js` file.
+ But if you add it then don't forget to replace ``<username>`` with your username.
+
+Remove the ``~/rocket.chat-user-setup.js`` file
+
+::
+
+ [isabell@stardust ~]$ rm ~/rocket.chat-user-setup.js
+ [isabell@stardust ~]$
+
+Setup daemon
+------------
+
+Create ``~/etc/services.d/rocket.chat.ini`` with the following content (don't forget to use your URL for ``ROOT_URL``):
+
+.. code-block:: ini
+
+ [program:rocket.chat]
+ command=node %(ENV_HOME)s/rocket.chat/main.js
+ environment=MONGO_URL="mongodb://<username>_rocketchat:<password>@localhost:27017/rocketchat?replicaSet=rs01&authSource=admin",MONGO_OPLOG_URL="mongodb://<username>_rocketchat:<password>@localhost:27017/local?replicaSet=rs01&authSource=admin",ROOT_URL="https://<username>.uber.space/",PORT=3000
```

Comment:
How about using `%(ENV_USER)s_rocketchat` here to reduce the number of things to manually replace?